src/lib/gadgets/gadgets.ts

```diff
@@ -33,4 +34,27 @@ const Gadgets = {
   rangeCheck64(x: Field) {
     return rangeCheck64(x);
   },
+
+  /**
+   * Bitwise XOR gadget on {@link Field} elements. Equivalent to the [bitwise XOR `^` operator in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_XOR).
+   * A XOR gate works by comparing two bits and returning `1` if two bits differ, and `0` if two bits are equal.
+   *
+   * The `length` parameter lets you define how many bits should be compared. The output is not constrained to the length.
```

Comment:
"The output is not constrained to the length" is interesting. Because further below you write that the inputs need to be of that bit length. And the `xor` of two n-bit values is again an n-bit value. So, what is happening here?

The following might be the correct answer (but I don't know and you should check @Trivo25):
"The gate does not constrain the inputs to `length` bits and if they don't fit in `length` bits, the method fails to properly constrain the operation. However, if the inputs do fit in `length` bits, the output is guaranteed to do so as well"

Reply:
Yea thats right, I believe we had this discussion during reviewing the API PR, which I just took the doc comment from - let me know if the explanation makes more sense now